tests/prepare_test_data.py

```diff
@@ -0,0 +1,173 @@
+        basepath = os.path.join(destdir, project['name'])
+        try:
+            os.makedirs(basepath)
+        except OSError:
+            logger.error('Cannot create directory "{}"'.format(basepath))
+            continue
+        for commit in project['commits']:
+            logger.info('Processing commit "{}"'.format(commit))
+            path = os.path.join(basepath, commit)
+            data = {
+                'source_code_directory': path,
+                'directories_to_skip': project['skip-dirs'],
+                'project': project['name'],
+                'commit': commit,
+                'ipprefix': project['name'],
+            }
+            datafile = '{}.json'.format(path)
+            try:
+                with open(datafile, 'wb') as f:
+                    json.dump(data, f)
+            except IOError:
+                logger.error('Cannot open file "{}"'.format(datafile))
+                continue
+            try:
+                repo = git.Repo.clone_from(project['clone-url'], path)
+            except git.exc.GitCommandError:
+                logger.error('Cannot clone git repository "{}" to "{}"'.format(
+                    project['clone-url'], path))
+                continue
+            try:
+                archive = '{}.tar.gz'.format(path)
+                prefix = os.path.join(commit, '')
+                try:
+                    with open(archive, 'wb') as f:
+                        repo.archive(f, commit, prefix, format='tar.gz')
+                except (IOError, git.exc.GitCommandError):
+                    logger.error('Cannot create archive "{}"'.format(archive))
+                    continue
+                try:
+                    plugin = GoSymbolExtractor()
+                    if not plugin.setData(data):
+                        logger.error('Failed to set data to ' +
+                            'GoSymbolExtractor plugin')
+                        continue
+                    if not plugin.execute():
+                        logger.error('Failed to execute ' +
+                            'GoSymbolExtractor plugin')
+                        continue
+                    output_data = plugin.getData()
+                    if not output_data:
+                        logger.error('Failed to get data from ' +
+                            'GoSymbolExtractor plugin')
+                        continue
+                except:
+                    logger.error('Unhandled exception occured during ' +
+                        'execution of GoSymbolExtractor plugin')
+                    continue
+                api = '{}.api'.format(path)
+                try:
+                    with open(api, 'wb') as f:
+                        json.dump(output_data[1], f)
+                except IOError:
+                    logger.error('Cannot open file "{}"'.format(api))
+                    continue
+            finally:
+                shutil.rmtree(path, ignore_errors=True)
```

Comment:
I know, it's still there :-) But if I remove this, what's the point of making the archive?